addons/website_sale/models/product_product.py

```diff
@@ -90,3 +90,11 @@ def _is_sold_out(self):
     def _is_add_to_cart_allowed(self):
         self.ensure_one()
         return self.user_has_groups('base.group_system') or (self.sale_ok and self.website_published)
+
+    def _get_contextual_price_tax_selection(self):
+        self.ensure_one()
+        price = self._get_contextual_price()
+        line_tax_type = self.env['ir.config_parameter'].sudo().get_param('account.show_line_subtotals_tax_selection')
+        if line_tax_type == "tax_included" and self.taxes_id:
+            price = self.taxes_id.compute_all(price, product=self, partner=self.env['res.partner'])['total_included']
+        return price
```

Comment:
Why are you creating a whole new method instead of modifying `_get_contextual_price`

Reply:
I saw that _get_contextual_price was used in other xml and .py files and I didn't want to create bugs in them with my fix

Reply:
Yes indeed but they most likely will have the same issues, you also have to take into account that it could be a rental price in some cases

Reply:
So modifying _get_contextual_price will breack the current lines in event will break the following lines where price_reduce is computed without taxes using _get_contextual_price and price_reduce_taxinc uses price_reduce to add taxes
https://github.com/odoo/odoo/blob/9731ed94c86baf1539ac9b5031c3833bcff6f2c6/addons/event_booth_sale/models/event_booth_category.py#L47-L68 I'm still looking but to avoid breaking (or having to change) that we might need to replace other _get_contextual_price by _get_contextual_price_tax_selection rather than modifying _get_contextual_price

Reply:
The two other references I found are not an issue: https://github.com/odoo/odoo/blob/9e99a9df464d97a74ca320d200599f7dce2d3050/addons/sale_product_configurator/views/templates.xml#L138 is hidden and its value is overwritten directly at https://github.com/odoo/odoo/blob/b7429b34ad4cac14f4afcc1518cec612c8a99f4a/addons/sale/static/src/js/variant_mixin.js#L530-L534 
and https://github.com/odoo/odoo/blob/9e99a9df464d97a74ca320d200599f7dce2d3050/addons/website_sale/static/src/xml/website_sale_utils.xml#L30 seems not to be used anymore (or at least I didn't found it anywhere)